tools/create_release.py

```diff
@@ -120,17 +121,28 @@ def find_upload_url(self) -> None:
         response.raise_for_status()
         for r in response.json():
             if r['tag_name'] == self.tag:
-                self.upload_url = r['upload_url'].replace('{?name,label}','')
-                print('Found release:', self.upload_url)
-                return
+                if r['draft']:
+                    delete_api = f'https://api.github.com/repos/{self.repo}/releases/{r["id"]}'
+                    response = requests.delete(delete_api, headers=headers)
+                    response.raise_for_status()
+                    print('Deleted stale release draft:', r['id'])
+                else:
+                    raise Exception('Refusing to recreate existing release')
 
+        commit = subprocess.check_output(
+            ['git', 'rev-parse', 'HEAD'], text=True
+        ).strip()
```

Comment:
Preferably function calls in python don't close on their own line at the same indentation level as they started. This is strictly avoided in meson itself. Unlike say dicts, it's not really idiomatic to do this in a language with significant indentation.

I'd usually try to fit it all on one line if possible. You could also define `cmd = []` on one line and reuse it on the next line and still have code that is strictly more compact and readable.